pkg/apis/devconsole/v1alpha1/component_types.go

```diff
@@ -36,8 +36,39 @@ type Component struct {
 	Status ComponentStatus `json:"status,omitempty"`
 }
 
-func (c *Component) GetName() string {
-	return c.Name
+// GetLabelName retrieves the name label. If not present,
+// infer the name by using the component name value.
+func (c *Component) GetLabelName() string {
+	name := c.Labels["app.kubernetes.io/name"]
+	if name == "" {
+		name = c.Name
+	}
+	return name
+}
+
+// GetLabelComponent retrieves the component label.
+func (c *Component) GetLabelComponent() string {
+	return c.Labels["app.kubernetes.io/component"]
```

Comment:
@tinakurian what if the component name is empty?

Reply:
@DhritiShikhar This is handled in the other PR in devconsole-operator. If it is empty, it will not be added to the map which contains labels. See: https://github.com/redhat-developer/devconsole-operator/pull/114/files#diff-f63bd964e50942ecf39aa15712e6361dR24

Reply:
Could we move this to a different .go file?

Reply:
@sbose78 sounds good, where would you like me to move it to?

Reply:
How about .._types_labels.go ?

Reply:
done!